lib/pcr.c

```diff
@@ -564,9 +565,9 @@ tool_rc pcr_read_pcr_values(ESYS_CONTEXT *esys_context,
         TPML_DIGEST *v;
         tool_rc rc = tpm2_pcr_read(esys_context, ESYS_TR_NONE, ESYS_TR_NONE,
                 ESYS_TR_NONE, &pcr_selection_tmp, &pcr_update_counter,
-                &pcr_selection_out, &v);
+                &pcr_selection_out, &v, cp_hash, parameter_hash_algorithm);
 
-        if (rc != tool_rc_success) {
+        if (rc != tool_rc_success || cp_hash->size) {
```

Comment:
```suggestion
        if (rc != tool_rc_success || (cp_hash && cp_hash->size)) {
```

Reply:
we also need to test if cp_hash is valid, else we'd accidentally dereference a null pointer.